### ISSUE TITLE
Chrome41: "The callback provided as parameter 1 is not a function."

### ISSUE THREAD

**Issue by daseinhorn** [mozilla/localForage#364]
Since updating to the latest version of the chrome browser, I get errors in my log like this:

```shell
TypeError: Failed to execute 'transaction' on 'Database': The callback provided as parameter 1 is not a function.
```

It is thrown at this line (localforage.js:1249):

```javascript
var store = dbInfo.db.transaction(dbInfo.storeName, 'readonly')
                              .objectStore(dbInfo.storeName);
```

Is there anything I can do about that?

Thanks,
daseinhorn

**Comment by tofumatt** [comment]
Is this Chrome stable? Which operating system and version of Chrome are you using? It should be working okay.

- tofumatt (Sent from mobile)

On 25 March 2015 14:06:33 CET, Das Einhorn <notifications@github.com> wrote:Since updating to the latest version of the chrome browser, I get errors in my log like this:  TypeError: Failed to execute 'transaction' on 'Database': The callback provided as parameter 1 is not a function.   It is thrown at this line (localforage.js:1249):  var store = dbInfo.db.transaction(dbInfo.storeName, 'readonly') .objectStore(dbInfo.storeName);   Is there anything I can do about that?  Thanks, daseinhorn  —Reply to this email directly or view it on GitHub.

**Comment by daseinhorn** [comment]
Sorry for the inaccurate issue description. 
I tried different browsers and the problem is only occurring since Chrome41. My exact Version: 41.0.2272.101 (64-bit) (stable).

The problem only occurs if the driver is set to WebSQL:
```javascript
localforage.config({
        name: 'MYName',
        version: 1.0,
        storeName: 'mynamestore',
        driver: localforage.WEBSQL
    });
```
If I change the driver to
```javascript
        driver: localforage.INDEXEDDB
```

everything works fine again.

I also tried the newest dev build (Chrome43) and the exact same issue occurs.

Thanks,
daseinhorn

**Comment by tofumatt** [comment]
It is troubling that WebSQL doesn’t work, though the default driver for Chrome should be IndexedDB. Still worth looking at this, but thanks for the update; less dire than I first thought. :-)

**Comment by daseinhorn** [comment]
Definitely. Since the IndexedDB driver works absolutely fine it really is not critical.
But strange enough that Chrome introduces breaking changes to the WebSQL API without a notice.

**Comment by tofumatt** [comment]
Yeah. Worse, we’ve seen similar things in the newest version of Firefox. :-(
